packages/console/src/mdx-components/Tabs/index.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Nullable } from '@silverhand/essentials';
+import React, { useState, isValidElement, type ReactElement, cloneElement } from 'react';
+
+import type { Props as TabItemProps } from '../TabItem';
+
+type Props = {
+  className?: string;
+  children: ReactElement<TabItemProps>;
+};
+
+// A very rough duck type, but good enough to guard against mistakes while
+// allowing customization
+function isTabItem(comp: ReactElement): comp is ReactElement<TabItemProps> {
+  return typeof comp.props.value !== 'undefined';
+}
+
+const Tabs = ({ className, children }: Props): JSX.Element => {
+  const verifiedChildren = React.Children.map(children, (child) => {
+    if (isValidElement(child) && isTabItem(child)) {
+      return child;
+    }
+  });
+
+  const values =
+    // Only pick keys that we recognize. MDX would inject some keys by default
+    verifiedChildren.map(({ props: { value, label } }) => ({
+      value,
+      label,
+    }));
+
+  const [selectedValue, setSelectedValue] = useState<string>();
+  const tabReferences: Array<Nullable<HTMLLIElement>> = [];
+
+  const handleTabChange = (
+    event: React.FocusEvent<HTMLLIElement> | React.MouseEvent<HTMLLIElement>
+  ) => {
+    const newTab = event.currentTarget;
+    const newTabIndex = tabReferences.indexOf(newTab);
+    const newTabValue = values[newTabIndex]?.value;
+
+    if (newTabValue !== selectedValue) {
+      setSelectedValue(newTabValue);
+    }
+  };
+
+  const handleKeydown = (event: React.KeyboardEvent<HTMLLIElement>) => {
+    // eslint-disable-next-line @silverhand/fp/no-let
+    let focusElement: Nullable<HTMLLIElement> = null;
+
+    switch (event.key) {
+      case 'ArrowRight': {
+        const nextTab = tabReferences.indexOf(event.currentTarget) + 1;
+        // eslint-disable-next-line @silverhand/fp/no-mutation
+        focusElement = tabReferences[nextTab] ?? tabReferences[0] ?? null;
+        break;
+      }
+
+      case 'ArrowLeft': {
+        const previousTab = tabReferences.indexOf(event.currentTarget) - 1;
+        // eslint-disable-next-line @silverhand/fp/no-mutation
+        focusElement =
+          tabReferences[previousTab] ?? tabReferences[tabReferences.length - 1] ?? null;
+        break;
+      }
+      default:
+        break;
+    }
+
+    focusElement?.focus();
```

Comment:
would be better to wrap the "get focusElement" logic into a function instead of mutation

Reply:
This `Tabs` component is copied from docusaurus source code. Although I made some simplification to make it work, I haven't touched this part.
There are some other to-do stuff for this component, too. Currently the tab navigation does not work, and the styles are also missing. Will create another task to improve this component.
https://linear.app/silverhand/issue/LOG-2331/improve-tabs-and-tabitem-components-in-mdx-components

Reply:
cool